### ISSUE TITLE
[augmentation-libraries] Missing grammar rule for extension and enum augmentations?

### ISSUE THREAD

**Issue by eernstg** [dart-lang/language#3657]
Thanks to @sgrekhov for bringing up this question. In the augmentation library feature specification, there is a grammar rule for augmenting class declarations: 

```ebnf
classDeclaration ::=
  'augment'? 'abstract'? 'class' identifier typeParameters?
  // Rest of rule...
```

Presumably there should be a similar rule for augmenting extension declarations and for augmenting enum declarations, but they do not seem to be included. Extension type declarations are not mentioned, but presumably they could be augmented just as well.

@jakemac53, should these grammar rules be added, or is there something deeper in this?

**Comment by jakemac53** [comment]
cc @munificent would you be able to tackle some of these grammar related issues?

**Comment by jakemac53** [comment]
> @jakemac53, should these grammar rules be added, or is there something deeper in this?

It just hasn't been written

**Comment by lrhn** [comment]
I think it should be possible to augment any declaration, except possibly type alias. (Although you could probably add annotations to them.)

**Comment by jakemac53** [comment]
I think in terms of the grammar if we can just allow it for all declarations that would definitely simplify things. There might not be anything you can actually _do_ with some of them though (like type aliases), but we could attach meaning in the future.

**Comment by eernstg** [comment]
See https://github.com/dart-lang/language/pull/3663 for a proposal.

**Comment by munificent** [comment]
Can this be closed now?

**Comment by munificent** [comment]
Fixed by #3663. Thanks, @eernstg!
